### PR TITLE
Q&A - Add an Answer

### PR DESCRIPTION
**Pull request by tyler-audio**
Component meets all requirements laid out in business document.
Styling for modal box is complete.
** may need additional styling done when individual question component is implemented

### CODE REVIEW

File: client/src/components/QuestionAnswer/AddAnswer.jsx [hunk 1]
```diff
@@ -1,7 +1,114 @@
...
+              id={`answer-modal-answer${props.i}`}
+              className="answer-modal-textarea answer-input"
+              name="body"
+              required
+              onInvalid={(e) => {
+                e.target.setCustomValidity('You must enter the following: Your Answer');
+              }}
+              maxLength="1000"
+            ></textarea>
+            <div id="answer-text-feedback">{}</div>
```
Comment (lawlorseanr) [inline client/src/components/QuestionAnswer/AddAnswer.jsx:73]: What is this for?

File: client/src/components/QuestionAnswer/QuestionComponent.jsx [hunk 2]
```diff
@@ -8,12 +8,14 @@ import AtelierAPI from '../../lib/atelierAPI';
...
 
+import AddAnswer from './AddAnswer.jsx';
+
 export default props => {
   const dispatch = useDispatch();
 
   const question = props.question;
   const answers = useSelector(state => state.answers[question.question_id]);
-  const product = useSelector(state => state.product.productId)
+  const product = useSelector(state => state.product.productId);
```
Comment (lawlorseanr) [inline client/src/components/QuestionAnswer/QuestionComponent.jsx:18]:
```suggestion
  const state = useSelector(state => {
    product: state.product.productId,
    answers: state.answers[question.question_id]
  });
```

File: client/src/components/QuestionAnswer/QuestionComponent.jsx [hunk 3]
```diff
@@ -94,7 +96,8 @@ export default props => {
             })
               .catch(err => console.error(err));
               e.target.innerHTML = 'Reported';
-          }}>Report</span>
+          }}>Report</span> */}
```
Comment (lawlorseanr) [inline client/src/components/QuestionAnswer/QuestionComponent.jsx:99]: these should be removed

File: client/src/components/QuestionAnswer/QuestionComponent.jsx [hunk 4]
```diff
@@ -80,11 +82,11 @@ export default props => {
                 helpfulness: question.question_helpfulness + 1
               }
             })
-              .then(res => actions.getQuestions(dispatch, product, 1, 4))
+              .then(res => {})
```
Comment (lawlorseanr) [inline client/src/components/QuestionAnswer/QuestionComponent.jsx:85]: this can be removed

**Review by lawlorseanr — APPROVED** [review]
suggestions included, but nothing that would prevent merge

File: client/src/components/QuestionAnswer/AddAnswer.jsx (hunk 1 above)
Comment (tyler-audio) [reply client/src/components/QuestionAnswer/AddAnswer.jsx:73]: oh yeah, i forgot to remove that. it isn't doing anything at all

File: client/src/components/QuestionAnswer/QuestionComponent.jsx (hunk 2 above)
Comment (tyler-audio) [reply client/src/components/QuestionAnswer/QuestionComponent.jsx:18]: wouldn't this require more refactoring than just this one line?

File: client/src/components/QuestionAnswer/QuestionComponent.jsx (hunk 3 above)
Comment (tyler-audio) [reply client/src/components/QuestionAnswer/QuestionComponent.jsx:99]: Should mention that these changes in this QuestionComponent file aren't pertinent to the add an answer feature. I have a separate branch for the individual questions component, i believe some of this was just done when I was trying to solve the issue you took a look at  for me the other day. This report portion in particular was commented out so i wouldn't be tempted to click it and remove the questions.

**Pull request merged**